File: 119_Pascal'sTriangleII.py
class Solution(object):
    def getRow(self, rowIndex):
        """
        Given an index k, return the kth row of the Pascal's triangle.
        For example, given k = 3,
        Return [1,3,3,1].
        Note: Could you optimize your algorithm to use only O(k) extra space?
        :type rowIndex: int
        :rtype: List[int]
        """
        if rowIndex == 0:
            return [1]
        elif rowIndex == 1:
            return [1,1]
        r = [1, 1]
        for i in xrange(rowIndex - 1):
            for j in xrange(len(r) - 1):
                r[j] = r[j] + r[j + 1]
            r.insert(0, 1)
        
        return r

File: 119.Pascal'sTriangleII.java
import java.util.ArrayList;

class Solution {
    public List<Integer> getRow(int rowIndex) {
        ArrayList<Integer> result = new ArrayList<Integer>();
        for (int i = 0; i < rowIndex + 1; i += 1) {
            result.add(0, 1);
            for (int j = 1; j < result.size() - 1; j += 1) {
                result.set(j, result.get(j) + result.get(j + 1));
            }
        }
        return result;
    }
}